pkg/op/op.go

```diff
@@ -229,6 +229,11 @@ func (o *openidProvider) GrantTypeJWTAuthorizationSupported() bool {
 	return true
 }
 
+func (o *openidProvider) GrantTypeClientCredentialsSupported() bool {
+	_, ok := o.storage.(ClientCredentialsXXX)
```

Comment:
if the storage provider implements _ClientCredentialsXXX_ then the grant type is supported.